PixArt/models/PixArt_blocks.py

```diff
@@ -193,9 +216,17 @@ def forward(self, x, mask=None, HW=None, block_id=None):
             )
         else:
             q, k, v = map(lambda t: t.transpose(1, 2),(q, k, v),)
-            x = torch.nn.functional.scaled_dot_product_attention(
+
+            p = getattr(self.attn_drop, "p", 0) # IPEX.optimize() will turn attn_drop into an Identity()
+            
+            if reduce(operator.mul, q.size()) > Q_4GB_LIMIT and sdpa_32b:
```

Comment:
I think a safer bet would be checking the tensor size with `q.element_size() * q.nelement()`, since that accounts for dtype. Probably will have to adjust the limit as well. 

Also maybe doing the sdpa_32b none check first so it doesn't get calculated w/o intel though the performance is probably the same.

```py
if sdpa_32b is not None and (q.element_size() * q.nelement()) > Q_4GB_LIMIT:
```